### PR TITLE
3866 add support for registering new cce category cold rooms and freezer rooms

### LINKED ISSUE #3866: Add support for registering new CCE Category 'Cold rooms and freezer rooms'
| User story  | A.1.2 |
| :------------- | :------------- |
As a… | Health Care Worker (HCW) or warehouse worker
I want… | To be able to easily add my CCE to the system
So that… | I don’t need to type in a lot of data manually when registering my facility’s CCE 

**Solution A.1.2**
The system will allow users to register their CCE using a barcode scanner. 

The scanner will work with UNICEF’s proposed 2D DataMatrix format (see ‘Perf spec E003 BC01.1 Global asset identification - Proposed revision Nov 23’ PDF) by detecting the PQS product number in the 2D DataMatrix and importing the remaining data from the corresponding product entry in the WHO PQS Catalogue. 

The serial number will also be imported from the 2D DataMatrix. 

Further support for the 2D DataMatrix may be added as necessary once the proposed standard has been revised, detailed and approved by WHO PQS.

Users will also be able to easily search for and select my make/model of CCE from a list in the event that they can not / do not want to use the barcode scanning functionality. 

The following data will be captured for each WHO PQS CCE:

- E001 - Cold rooms and freezer rooms
  - Supplied by current WHO PQS Catalogue 
    - PQS code
    - Type
    - Class
    - Category
    - Manufacturer
    - Model
    - Energy source
    - Refrigerant type(s)
  - Not supplied by current WHO PQS Catalogue
    - Can be input manually by user
      - Asset number (country’s unique asset identifier)
      - Installation date
    - Can be retrieved from 2D DataMatrix or input manually by user
      - Serial number
      - External dimensions - WxDxH (in cm)
      - Storage capacity +5 °C (in litres)
      - Storage capacity -20 °C (in litres)
      - Storage capacity -70 °C (in litres)
      - Waterpack storage capacity (in Kg)
      - Waterpack freezing capacity per 24 hours (in Kg)
      - Energy consumption (stable running, continuous power) (in KW per day)
      - Energy consumption during freezing (in KW per day)
      - Hold over time (hours)
      - Climate zone
      - Freeze protection 
      - Temperature monitoring device (integrated, external or none)
      - Voltage stabilizer (integrated, external or none)
      - Warranty period (start-end date) 
      - Links to: 
        - PQS Specification sheet
        - Manufacturer’s Operating Manual
        - Manufacturer’s Warranty
        - Manufacturer’s Maintenance Instructions

---

| User story  | A.1.2.1 |
| :------------- | :------------- |
As a… | National EPI manager
I want… | To be able to specify the volume for each temperature range for any CCE that I add to a Sub catalogue - e.g. a non-WHO PQS Sub catalogue 
So that… | The volumes are available for use in cold storage capacity and demand calculations when the CCE is added by HCW to their CCE registry.  

**Solution A.1.2.1**
The system will allow central users to specify a volume when adding / importing new Sub catalogue items in the Categories ‘Refrigerators and freezers’ or ‘Cold rooms and freezer rooms’.

---
| User story  | A.1.2.2 |
| :------------- | :------------- |
As a… | Health Care Worker (HCW) or warehouse worker
I want… | To be able to specify the volume for each temperature range for any WHO PQS CCE that does not already include this information by default when I add it to my registry - e.g. E001 cold rooms that are custom built
So that… | The volumes are available for use in cold storage capacity and demand calculations when the CCE is added by HCW to their CCE registry.  

**Solution A.1.2.2**
The system will allow users to specify a volume when registering CCE in the WHO PQS Category E001 - ‘Cold rooms and freezer rooms’.

---

### LINKED ISSUE #3929: Capture additional data fields for registering CCE Category 'Insulated containers'
| User story  | A.1.2 |
| :------------- | :------------- |
As a… | Health Care Worker (HCW) or warehouse worker
I want… | To be able to easily add my CCE to the system
So that… | I don’t need to type in a lot of data manually when registering my facility’s CCE 

**Solution A.1.2**
The system will allow users to register their CCE using a barcode scanner. 

The scanner will work with UNICEF’s proposed 2D DataMatrix format (see ‘Perf spec E003 BC01.1 Global asset identification - Proposed revision Nov 23’ PDF) by detecting the PQS product number in the 2D DataMatrix and importing the remaining data from the corresponding product entry in the WHO PQS Catalogue. 

The serial number will also be imported from the 2D DataMatrix. 

Further support for the 2D DataMatrix may be added as necessary once the proposed standard has been revised, detailed and approved by WHO PQS.

Users will also be able to easily search for and select my make/model of CCE from a list in the event that they can not / do not want to use the barcode scanning functionality. 

The following data will be captured for each WHO PQS CCE:

- E004 - Insulated containers
  - Supplied by current WHO PQS Catalogue 
    - PQS code
    - Type
    - Class
    - Category
    - Manufacturer
    - Model
    - External dimensions - WxDxH (in cm)
  - Not supplied by current WHO PQS Catalogue
    - Can be input manually by user
      - Asset number (country’s unique asset identifier)
      - Installation date
    - Can be retrieved from 2D DataMatrix or input manually by user
      - Serial number
      - Warranty period (start-end date) 
      - Links to: 
        - PQS Specification sheet
        - Manufacturer’s Operating Manual
        - Manufacturer’s Warranty
        - Manufacturer’s Maintenance Instructions

### PR DESCRIPTION
**Pull request by jmbrunskill**
Fixes #3866
Fixes #3929
Fixes #3866

# 👩🏻‍💻 What does this PR do?

Adds the asset properties that can be entered for Cold Rooms & Fridge Freezers

## 💌 Any notes for the reviewer?

For the catalogue items we also need to add new properties but thinking I might leave them till after re-factoring those? 

For Cold Rooms we need 

- Energy source
- Refrigerant type(s)
- 
See
- [ ] #3971
- [ ] #3973 

# 🧪 Testing

- [ ] Create freezers and cold room using all the new asset catalogue items (E001/SOMETHING)
- [ ] Check the correct properties show up in the summary page

# 📃 Documentation

- [X] **Part of an epic**: documentation will be completed for the feature as a whole

### CODE REVIEW

File: server/repository/src/migrations/v2_01_00/assets/asset_property_data.rs [hunk 1]
```diff
@@ -1,12 +1,60 @@
...
+        INSERT INTO asset_property (id, key, name, value_type, allowed_values, asset_class_id, asset_category_id) VALUES ('freeze_protection', 'freeze_protection', 'Freeze protection', 'STRING', NULL, 'fad280b6-8384-41af-84cf-c7b6b4526ef0','7db32eb6-5929-4dd1-a5e9-01e36baa73ad');
+        INSERT INTO asset_property (id, key, name, value_type, allowed_values, asset_class_id, asset_category_id) VALUES ('temperature_monitoring_device', 'temperature_monitoring_device', 'Temperature monitoring device (integrated, external or none)', 'STRING', "integrated, external, none", 'fad280b6-8384-41af-84cf-c7b6b4526ef0','7db32eb6-5929-4dd1-a5e9-01e36baa73ad');
+        INSERT INTO asset_property (id, key, name, value_type, allowed_values, asset_class_id, asset_category_id) VALUES ('voltage_stabilizer', 'voltage_stabilizer', 'Voltage stabilizer (integrated, external or none)', 'STRING', "integrated, external, none", 'fad280b6-8384-41af-84cf-c7b6b4526ef0','7db32eb6-5929-4dd1-a5e9-01e36baa73ad');
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply server/repository/src/migrations/v2_01_00/assets/asset_property_data.rs:42]: Actually, while at it - I would capitalise the `Integrated,External,None` values. Do they need to be lowercase?

**Review by mark-prins — CHANGES_REQUESTED** [review]
Working nicely! thanks - just had a postgres thing to change.

Also wondered if we can we make the property name wider?

image

**Comment by adamdewey** [comment]
I think we don't need the 'Integrated', 'External' or 'None' in the title of the property - should be fairly evident from the dropdown?

File: server/repository/src/migrations/v2_01_00/assets/asset_property_data.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmbrunskill) [reply server/repository/src/migrations/v2_01_00/assets/asset_property_data.rs:42]: Yeah agreed, I fixed this on another branch but obviously forgot to run postgres tests here 🤦‍♂️

**Review by mark-prins — APPROVED** [review]
🤩  thanks for changes!

**Pull request merged**